src/Nomad.Sorter.Infrastructure/Cosmos/ChangeFeed/Processors/ParcelChangeFeedProcessor.cs

```diff
@@ -0,0 +1,30 @@
+using System.Security.AccessControl;
+using Microsoft.Azure.CosmosRepository;
+using Microsoft.Extensions.Logging;
+using Nomad.Sorter.Domain.Entities;
+using Nomad.Sorter.Domain.Enums;
+using Nomad.Sorter.Infrastructure.Cosmos.Items;
+
+namespace Nomad.Sorter.Infrastructure.Cosmos.ChangeFeed.Processors;
+
+public class ParcelChangeFeedProcessor : IChangeFeedItemProcessor<Parcel>
+{
+    private readonly ILogger<ParcelChangeFeedProcessor> _logger;
+    private readonly IRepository<ParcelIdLookup> _lookupRepository;
+
+    public ParcelChangeFeedProcessor(ILogger<ParcelChangeFeedProcessor> logger, IRepository<ParcelIdLookup> lookupRepository)
+    {
+        _logger = logger;
+        _lookupRepository = lookupRepository;
+    }
+    
+    public async ValueTask HandleAsync(Parcel parcel, CancellationToken cancellationToken)
+    {
+        _logger.LogInformation("Parcel change being processed for parcel with {ID}", parcel.ParcelId);
+        
+        if (parcel.Status is ParcelStatus.PreAdvice)
+        {
+            await _lookupRepository.UpdateAsync(parcel.ToParcelIdLookup(), cancellationToken);
```

Comment:
Confused by this method name, `ToParcelIdLookup`. Is this the Parcel Id used for lookup, or the Parcel lookup Id? Right now, it reads as the Parcel Id for lookup, but I'd imagine it's the Parcel lookup Id. 😄

Reply:
So this is another `Item` that allow for a point read to of the parcel by its ID to then hop to the proper parcel which is pk'd by the `deliveryRegionId` maybe the name is a little confusing I will have a think 🤔